dashboard/test/ui/step_definitions/steps.rb

```diff
@@ -131,9 +131,20 @@ def install_js_error_recorder
   wait_with_timeout.until { @browser.execute_script("return $(#{selector.dump}).text();").include? text }
 end
 
-When /^I wait until element "([^"]*)" is visible$/ do |selector|
+def jquery_is_element_visible(selector)
+  "return $(#{selector.dump}).is(':visible') && $(#{selector.dump}).css('visibility') !== 'hidden';"
```

Comment:
We have an inconsistent definition of "visible" spread around our UI tests; we still do, somewhat, but I clean it up quite a bit here.  In many places we were only checking `$(selector).is(':visible')`, but in a few others we also checked `$(selector).css('visibility') !== 'hidden'`; I guess the one isn't covered by the other.  This led me to the odd discovery that I could "Wait until dialog is visible" then assert "dialog is visible" and fail on the second step :sob:.

The changes I've made prefer checking both where we used to check one, and tests are passing with that change.